requests/adapters.py

```diff
@@ -284,7 +302,13 @@ def request_url(self, request, proxies):
         """
         proxy = select_proxy(request.url, proxies)
         scheme = urlparse(request.url).scheme
-        if proxy and scheme != 'https':
+        if proxy:
+            proxy_scheme = urlparse(proxy).scheme
+
+        if proxy and proxy_scheme.lower().startswith('socks'):
```

Comment:
You conditionally create `proxy_scheme` but unconditionally reference it here unless my coffee hasn't kicked in. Shouldn't this block look like:

```py
proxy_scheme = ''
if proxy:
    proxy_scheme = urlparse(proxy).scheme

if proxy and proxy_scheme.lower().startswith('socks'):
    url = request.path_url
# ...
```